src/functions/utils.php

```diff
@@ -338,4 +338,53 @@ function tribe_retrieve_object_by_hook( $class_name, $hook, $priority ) {
 	function tribe_is_wpml_active() {
 		return ( class_exists( 'SitePress' ) && defined( 'ICL_PLUGIN_PATH' ) );
 	}
-}
\ No newline at end of file
+}
+
+if ( ! function_exists( 'tribe_post_exists' ) ) {
+	/**
+	 * Checks if a post, optionally of a specific type, exists in the database.
+	 *
+	 * This is a low-level database check that will ignore caches and will
+	 * check if there is an entry, in the posts table, for the post.
+	 *
+	 * @since TBD
+	 *
+	 * @param string|int $post_id_or_name Either a post ID or a post name.
+	 * @param null       $post_type       An optional post type.
+	 *
+	 * @return bool|int The matching post ID if found, `false` otherwise
+	 */
+	function tribe_post_exists( $post_id_or_name, $post_type = null ) {
+		if ( $post_id_or_name instanceof WP_Post ) {
+			$post_id_or_name = $post_id_or_name->ID;
+		}
+
+		global $wpdb;
+
+		$query_template = "SELECT ID FROM {$wpdb->posts} WHERE %s";
+		$query_vars     = array();
+		$where          = '';
+
+		if ( is_numeric( $post_id_or_name ) ) {
+			$where        = 'ID = %d';
+			$query_vars[] = $post_id_or_name;
+		} elseif ( is_string( $post_id_or_name ) ) {
+			$where        = 'post_name = %s';
+			$query_vars[] = $post_id_or_name;
+		}
+
+		if ( null !== $post_type ) {
```

Comment:
Could `is_string()` be a stronger and safer test (what if, glossing over the docblock, someone passes in an array or some other unexpected but non-null type)?